### PR TITLE
Hw/minio setup

### PR DESCRIPTION
**Pull request by opetliak**

### CODE REVIEW

File: src/utils/minio_crud.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        self.client = Minio(endpoint, access_key=access_key, secret_key=secret_key, secure=secure)
+
+    def create_bucket(self, bucket_name):
+        try:
+            if not self.client.bucket_exists(bucket_name):
+                self.client.make_bucket(bucket_name)
+                
+        except S3Error as exc:
+            return False
+        return True
```
Comment (truskovskiyk) [inline src/utils/minio_crud.py]: why do we need to return the bool value here? if got an error - let's raise it too

File: src/utils/minio_crud.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    def list_buckets(self):
+        return self.client.list_buckets()
+
+    def write_file(self, bucket_name, object_name, file_path):
+        try:
+            self.client.fput_object(bucket_name, object_name, file_path)
+        except S3Error as exc:
+            print(f"Error writing file: {exc}")
+            return False
+        return True
```
Comment (truskovskiyk) [inline src/utils/minio_crud.py]: not a big fan of this design pattern)

File: tests/utils/test_minio_crud_client.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import sys
+import os
+import io
+import time
+
+from pathlib import Path
+
+project_root = str(Path(__file__).resolve().parent.parent.parent)
+sys.path.append(project_root)
```
Comment (truskovskiyk) [inline tests/utils/test_minio_crud_client.py]: it's better to use PYTHONPATH env variable instead 
https://bic-berkeley.github.io/psych-214-fall-2016/using_pythonpath.html

File: tests/utils/test_minio_crud_client.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+from src.utils.minio_crud import MinioCRUDClient
+import numpy as np
+import pytest
+from minio.error import S3Error
+from PIL import Image
+
+
+ENDPOINT = os.environ.get("MINIO_ENDPOINT", "192.168.49.2:32000")
+ACCESS_KEY = "3JnbRlFAACF1v5YG"
+SECRET_KEY = "lYFakKmgSpQbcKREnjPid3CXkujxPbJ8"
```
Comment (truskovskiyk) [inline tests/utils/test_minio_crud_client.py]: please don't commit creds to the code, use env variables

**Review by truskovskiyk — COMMENTED** [review]
great work, please see feedback

File: src/utils/minio_crud.py (hunk 1 above)
Comment (opetliak) [reply src/utils/minio_crud.py]: updated

File: src/utils/minio_crud.py (hunk 2 above)
Comment (opetliak) [reply src/utils/minio_crud.py]: updated but let me know if there is a better pattern

File: tests/utils/test_minio_crud_client.py (hunk 3 above)
Comment (opetliak) [reply tests/utils/test_minio_crud_client.py]: updated

File: tests/utils/test_minio_crud_client.py (hunk 4 above)
Comment (opetliak) [reply tests/utils/test_minio_crud_client.py]: updated

Comment (truskovskiyk) [reply tests/utils/test_minio_crud_client.py]: thanks!

File: tests/utils/test_minio_crud_client.py (hunk 3 above)
Comment (truskovskiyk) [reply tests/utils/test_minio_crud_client.py]: great

File: src/utils/minio_crud.py (hunk 2 above)
Comment (truskovskiyk) [reply src/utils/minio_crud.py]: usually, if you check something and return bool after, it's just easy to return the check itself : 

like

```
if cond:
   return True
return False 
``` 

should be just 

```
return cond
```

File: src/utils/minio_crud.py (hunk 1 above)
Comment (truskovskiyk) [reply src/utils/minio_crud.py]: thanks!

**Review by truskovskiyk — APPROVED** [review]

**Pull request merged**